File: day6/app/src/main/java/fan/akua/day6/activities/DetailActivity.java
package fan.akua.day6.activities;

import android.app.Activity;
import android.content.Intent;
import android.media.Image;
import android.os.Bundle;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.ToggleButton;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.app.ActivityOptionsCompat;

import com.bumptech.glide.Glide;

import org.greenrobot.eventbus.EventBus;

import java.io.Serializable;

import fan.akua.day6.R;
import fan.akua.day6.bean.MultiBean;
import fan.akua.day6.events.StarEvent;

public class DetailActivity extends AppCompatActivity {
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        MultiBean dat = (MultiBean) getIntent().getSerializableExtra("dat");
        int position = getIntent().getIntExtra("pos", 0);
        setContentView(R.layout.activity_detail);
        ImageView imageView = findViewById(R.id.img);
        TextView tv = findViewById(R.id.tv);
        ToggleButton toggleButton = findViewById(R.id.star);

        toggleButton.setBackgroundResource(dat.isStated() ? R.drawable.baseline_star_24 : R.drawable.baseline_star_border_24);
        toggleButton.setOnCheckedChangeListener((buttonView, isChecked) -> {
            dat.setStated(isChecked);
            EventBus.getDefault().post(new StarEvent(dat, position));
            toggleButton.setBackgroundResource(isChecked ? R.drawable.baseline_star_24 : R.drawable.baseline_star_border_24);
        });

        if (dat.getItemType() == MultiBean.Type.Txt.ordinal()) {
            tv.setText(dat.getTxt());
        } else if (dat.getItemType() == MultiBean.Type.Img.ordinal()) {
            Glide.with(this).load(dat.getImgUrl()).into(imageView);
        } else {
            throw new RuntimeException("what's wrong with you?!");
        }

    }
}

File: day11/app/src/main/kotlin/fan/akua/day11/activities/DetailActivity.kt
package fan.akua.day11.activities

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.bumptech.glide.Glide
import fan.akua.day11.R
import fan.akua.day11.api.impl.LocalRepository
import fan.akua.day11.bean.ImageBean
import fan.akua.day11.bean.TextBean
import fan.akua.day11.databinding.ActivityDetailBinding
import fan.akua.day11.ktx.akuaEdgeToEdge
import fan.akua.day11.utils.RandomBlurTransformation


class DetailActivity : AppCompatActivity() {
    private lateinit var binding: ActivityDetailBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        akuaEdgeToEdge()
        super.onCreate(savedInstanceState)
        binding = ActivityDetailBinding.inflate(layoutInflater)
        setContentView(binding.root)
        val itemId = intent.getIntExtra("item_id", 0)

        val data = LocalRepository.getData().first {
            it.id == itemId
        }

        val type = intent.getIntExtra("item_type", 0)
        when (type) {
            1 -> {
                data as ImageBean
                Glide.with(this)
                    .load(data.resID)
                    .transform(RandomBlurTransformation(this))
                    .into(binding.img)
            }

            2 -> {
                data as TextBean
                binding.tv.text = data.string
            }
        }
        binding.star.isChecked = data.isStarred
        binding.star.setBackgroundResource(if (binding.star.isChecked) R.drawable.baseline_star_24 else R.drawable.baseline_star_border_24)
        binding.star.setOnCheckedChangeListener { _, isChecked ->
            LocalRepository.starOrUnstarData(itemId, true)
            binding.star.setBackgroundResource(if (isChecked) R.drawable.baseline_star_24 else R.drawable.baseline_star_border_24)
        }

    }
}